src/login/logind.c

```diff
@@ -984,7 +984,7 @@ static void manager_gc(Manager *m, bool drop_not_started) {
                 /* Normally, this should make the session referenced
                  * again, if it doesn't then let's get rid of it
                  * immediately */
-                if (!session_check_gc(session, drop_not_started)) {
+                if (!session_check_gc(session, drop_not_started) || session->stopping)  {
```

Comment:
nah, we shouldn't GC this if it's still stopping, but only when it fully stopped.

This tapes over another problem really: why is the session never fully stopped for you in the first place?

Reply:
OK, do you have some suggestion on how can I check why my sessions aren't fully stopping?

Reply:
> OK, do you have some suggestion on how can I check why my sessions aren't fully stopping?

See my comments here: https://github.com/systemd/systemd/issues/8015#issuecomment-361920691

i.e. "loginctl session-status" might help tracking this down

Reply:
cool, tnx!